### PR TITLE
Hackathon Improvements, Dec 2023

### PR DESCRIPTION
**Pull request by logyball**
Large-scale changes for `unused-exporter`! Outcome of the December 2023 Grafana hackathon team: Garbage Pail Kids aka Los Basuritos 

# What

A more comprehensive `unused-exporter`.  Adding a `total_size` metric as well as dramatically improving polling performance using caching.

# Why

Because it's fun!  Also so that we could make this something that actually runs as a service.

# Major Changes

- Authorization changes: all providers can now be authorized from environment vars
- Create a cache that allows prometheus metrics to be created on a pre-defined polling interval, rather than on-demand when scraped
  - This is so that these metrics, which seldom change, are not polled every 15s or so
- 

# Minor Changes

- Logger swapped out, now we use `slog`
- Added provider ID to the provider struct
- 

# Follow-ups

- Push docker image to a public repository?

### CODE REVIEW

**Review by inkel — COMMENTED** [review]
I did an initial review, and it looks good! Just left a change request and a comment, happy to discuss it before we merge this.

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -2,27 +2,37 @@ name: Go
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, dev/hackathon-dec-2023 ]
```
[the remark this answers is not in the document]
Comment (logyball) [reply .github/workflows/go.yml:5]: agreed

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,9 @@ BUILDFLAGS = -v
 BENCH = .
 BENCHFLAGS = -benchmem -bench=${BENCH}
 
+IMAGE_PREFIX ?= us.gcr.io/kubernetes-dev
+GIT_VERSION ?= $(shell git describe --tags --always --dirty)
```
[the remark this answers is not in the document]
Comment (logyball) [reply Makefile:8]: Agreed - do you think that this is a problem for this present PR, or could we address this in a follow up?  

This seems like something we'll need to do with some care, as we'll want to figure out how to push this publicly with auth, etc.

[the remark this answers is not in the document]
Comment (inkel) [reply Makefile:8]: Definitely out of the scope of this PR

**Review by inkel — APPROVED** [review]

**Review by inkel — APPROVED** [review]

**Review by rooneyshuman — DISMISSED** [review]
LGTM!

**Pull request merged**